vimiv/imutils/exif.py

```diff
@@ -10,139 +10,361 @@
-                    piexif.TYPES.Ascii,
-                    piexif.TYPES.Undefined,
-                ):  # byte encoded
-                    self[keyname] = val.decode()
-                elif keytype in (
-                    piexif.TYPES.Rational,
-                    piexif.TYPES.SRational,
-                ):  # (int, int) <=> numerator, denominator
-                    self[keyname] = f"{val[0]}/{val[1]}"
+
+        return exif
+
+    def copy_exif(self, dest: str, reset_orientation: bool = True) -> None:
+        def get_copy_args():
+            """Get filetype specific copy arguments."""
+
+            file_type = imghdr.what(dest).upper()
+            args = [False, False, False, False]
+
+            if file_type in [
+                "JPEG",
+                "JPG",
+                "EXV",
+                "CR2",
+                "CRW",
+                "MRW",
+                "TIFF",
+                "TIF",
+                "WEBP",
+                "DNG",
+                "NEF",
+                "PEF",
+                "ARW",
+                "RW2",
+                "SR2",
+                "SRW",
+                "ORF",
+                "PNG",
+                "PGF",
+                "RAF",
+                "PSD",
+                "JP2",
+            ]:
+                args[0] = True
+
+            if file_type in [
+                "JPEG",
+                "JPG",
+                "EXV",
+                "CR2",
+                "MRW",
+                "TIFF",
+                "TIF",
+                "DNG",
+                "NEF",
+                "PEF",
+                "ARW",
+                "RW2",
+                "SR2",
+                "SRW",
+                "ORF",
+                "PNG",
+                "PGF",
+                "RAF",
+                "PSD",
+                "JP2",
+            ]:
+                args[1] = True
+
+            if file_type in [
+                "JPEG",
+                "JPG",
+                "EXV",
+                "CR2",
+                "MRW",
+                "TIFF",
+                "TIF",
+                "WEBP",
+                "DNG",
+                "NEF",
+                "PEF",
+                "ARW",
+                "RW2",
+                "SR2",
+                "SRW",
+                "ORF",
+                "PNG",
+                "PGF",
+                "RAF",
+                "EPS",
+                "XMP",
+                "PSD",
+                "JP2",
+            ]:
+                args[2] = True
+
+            if file_type in ["JPEG", "JPG", "EXV", "CRW", "PNG", "PGF"]:
+                args[3] = True
+
+            return tuple(args)
+
+        if reset_orientation:
+            with contextlib.suppress(KeyError):
+                self._metadata["Exif.Image.Orientation"] = ExifOrientation.Normal
+        try:
+            dest_image = pyexiv2.ImageMetadata(dest)
+            dest_image.read()
+
+            self._metadata.copy(dest_image, *get_copy_args())
```

Comment:
I am a bit afraid of the behemoth I just came up with, but how about not checking the file type but just trying the different parts one-by-one and ignoring any errors (we could add a debug log instead of just suppressing if we opt for this):
```
    dest_image = pyexiv2.ImageMetadata(dest)
    dest_image.read()
    for copy_args in set(itertools.permutations((True, False, False, False))):
        with contextlib.suppress(ValueError):
            src_image.copy(dest_image, *copy_args)
    with contextlib.suppress(OSError):
        dest_image.write()
```

This seems rather pythonic as we ask for forgiveness instead of permission, but I am not happy with the fact that the pyexiv2 does not seem to provide a simple method to get the copy args we are manually creating/iterating.

Reply:
Oh wow, your solution is actually really smart. At first I did not get it and I already wanted to comment that this is not gonna work :blush:  But it is probably the clearest solution we can get.

It is indeed a bit unsatisfying that `py3exiv2` does not provide a more handy interface for copying metadata. But since this package is maintained, we could try to get this functionality into `py3exiv2` maybe.

Reply:
First of all, merry Christmas Jean-Claude, I hope you enjoy your holidays :christmas_tree:

Would you like to add this in then? Probably with some debug logging like:
```
try:
    ...
except OSError as e:
    _logger.debug("Failed to write exif data for '%s': ''%s'", dest, str(e)")
```

Reply:
Thank you, I hope you have had a merry Christmas too! :gift: 

I have added this method in a89d6fdba3b844b36457928852cce5ec13eda2ab

Reply:
Awesome, thank you!